R/teal_data-class.R

```diff
@@ -0,0 +1,88 @@
+setOldClass("JoinKeys")
+
+#' Reproducible data.
+#'
+#' Reproducible data container class. Inherits code tracking behavior from [`teal.code::qenv-class`].
+#'
+#' This class provides an isolated environment in which to store and process data with all code being recorded.
+#' The environment, code, data set names, and data joining keys are stored in their respective slots.
+#' These slots should never be accessed directly, use the provided get/set functions.
+#'
+#' As code is evaluated in `teal_data`, messages and warnings are stored in their respective slots.
+#' If errors are raised, a `qenv.error` object is returned.
+#'
+#' @name teal_data-class
+#' @rdname teal_data-class
+#'
+#' @slot env (`environment`) environment containing data sets and possibly auxiliary variables
+#'  Access variables with [get_var()] or [`[[`].
+#'  No setter provided. Evaluate code to add variables into `@env`.
+#' @slot code (`character`) representing code necessary to reproduce the contents of `@env`.
+#'  Access with [`get_code`].
+#'  No setter provided. Evaluate code to append code to the slot.
+#' @slot id (`integer`) random identifier assigned to each line of code in `@code`. Used internally.
+#' @slot warnings (`character`) the warnings output when evaluating the code.
+#' @slot warnings (`character`) warnings raised when evaluating code.
+#'  Access with [get_warnings()].
+#' @slot messages (`character`) messages raised when evaluating code.
+#' @slot join_keys (`JoinKeys`) object specifying joining keys for data sets in `@env`.
+#'  Access or modify with [get_join_keys()].
```

Comment:
```suggestion
#'  Access or modify with [`get_join_keys`].
```

Reply:
+1 for using () in function names

Reply:
The distinction between topic and function seems unclear to me. Is a function NOT a topic? What is a topic then? 

Here's my view:
The docs should read "Change datanames slot with datanames function."
One changes the slot with `datanames<object> <- <new_datanames>` not with `datanames()`. If you insist on the latter, you lose consistency with `[[`.

Look at help pages for `environment`, `get`, `assign`, etc. Hardly a `foo()` in sight.

Reply:
+ I always thought that table is an example, not a directive.

Reply:
Ok, so a topic would be a help page and one can be dedicated to a concept or a function or a group of functions or anything in between. Correct?